src/error.rs

```diff
@@ -290,93 +382,181 @@ impl Display for rustls_result {
-        AlertCertificateExpired => Error::AlertReceived(alert::CertificateExpired).fmt(f),
-        AlertCertificateUnknown => Error::AlertReceived(alert::CertificateUnknown).fmt(f),
-        AlertIllegalParameter => Error::AlertReceived(alert::IllegalParameter).fmt(f),
-        AlertUnknownCA => Error::AlertReceived(alert::UnknownCA).fmt(f),
-        AlertAccessDenied => Error::AlertReceived(alert::AccessDenied).fmt(f),
-        AlertDecodeError => Error::AlertReceived(alert::DecodeError).fmt(f),
-        AlertDecryptError => Error::AlertReceived(alert::DecryptError).fmt(f),
-        AlertExportRestriction => Error::AlertReceived(alert::ExportRestriction).fmt(f),
-        AlertProtocolVersion => Error::AlertReceived(alert::ProtocolVersion).fmt(f),
-        AlertInsufficientSecurity => Error::AlertReceived(alert::InsufficientSecurity).fmt(f),
-        AlertInternalError => Error::AlertReceived(alert::InternalError).fmt(f),
-        AlertInappropriateFallback => Error::AlertReceived(alert::InappropriateFallback).fmt(f),
-        AlertUserCanceled => Error::AlertReceived(alert::UserCanceled).fmt(f),
-        AlertNoRenegotiation => Error::AlertReceived(alert::NoRenegotiation).fmt(f),
-        AlertMissingExtension => Error::AlertReceived(alert::MissingExtension).fmt(f),
-        AlertUnsupportedExtension => Error::AlertReceived(alert::UnsupportedExtension).fmt(f),
-        AlertCertificateUnobtainable => Error::AlertReceived(alert::CertificateUnobtainable).fmt(f),
-        AlertUnrecognisedName => Error::AlertReceived(alert::UnrecognisedName).fmt(f),
-        AlertBadCertificateStatusResponse => {
-            Error::AlertReceived(alert::BadCertificateStatusResponse).fmt(f)
-        }
-        AlertBadCertificateHashValue => Error::AlertReceived(alert::BadCertificateHashValue).fmt(f),
-        AlertUnknownPSKIdentity => Error::AlertReceived(alert::UnknownPSKIdentity).fmt(f),
-        AlertCertificateRequired => Error::AlertReceived(alert::CertificateRequired).fmt(f),
-        AlertNoApplicationProtocol => Error::AlertReceived(alert::NoApplicationProtocol).fmt(f),
-        AlertUnknown => Error::AlertReceived(alert::Unknown(0)).fmt(f),
-
-        CertSCTMalformed => Error::InvalidSct(sct::MalformedSct).fmt(f),
-        CertSCTInvalidSignature => Error::InvalidSct(sct::InvalidSignature).fmt(f),
-        CertSCTTimestampInFuture => Error::InvalidSct(sct::TimestampInFuture).fmt(f),
-        CertSCTUnsupportedVersion => Error::InvalidSct(sct::UnsupportedSctVersion).fmt(f),
-        CertSCTUnknownLog => Error::InvalidSct(sct::UnknownLog).fmt(f),
+            // These variants are local to this glue layer.
+            rustls_result::Ok => write!(f, "OK"),
+            Io => write!(f, "I/O error"),
+            NullParameter => write!(f, "a parameter was NULL"),
+            InvalidDnsNameError => write!(
+                f,
+                "server name was malformed (not a valid hostname or IP address"
```

Comment:
```suggestion
                "server name was malformed (not a valid hostname or IP address)"
```

Reply:
Good catch, thanks! I added a fix and force pushed. LMK if you'd prefer to see separate review fix commits that get squashed at the end.

Reply:
> Generally speaking I prefer separate review fix commits, and always use squash merges. But for small stuff like this I'm not particularly picky.

:+1: 

> Also I recently learned a nice trick: In the "user force-pushed the branch branch..." message, "force-pushed" is clickable and will take you to a diff between the old branch and the force-pushed branch.

Oh hey, TIL. That is handy.